### ISSUE TITLE
"could not compile dependency :dns" error when running `mix ecto.create`

### ISSUE THREAD

**Issue by LuySeiwert** [albirrkarim/mozilla-hubs-installation-detailed#34]
# Environment
OS: Ubuntu 22.04
Erlang version: 23.3 installed via asdf (only possible thanks to this workaround https://github.com/asdf-vm/asdf-erlang/issues/247#issuecomment-1464580043)
Elixir version: 1.14.3-otp-23 installed also via asdf

# Steps to reproduce
Inside the reticulum repository
`mix deps.get`
`mix ecto.create`

# Expected Outcome
That it works.

# Actual outcome
`could not compile dependency :dns` error

![image](https://github.com/albirrkarim/mozilla-hubs-installation-detailed/assets/95661355/60759e11-5da2-4ac9-86fc-e816001722ed)

# Things I tried to resolve this issue
- googling the issue, I didn't find anything, there are other people having this type of error but never with this specific library or project
- looking at the closed issues in this repository, there are two similar issues #24 #18, I tried their fixes, they didn't work for me (still got the same error message both times)

Any idea why this is happening?

**Comment by albirrkarim** [comment]
I try to clone the latest reticulum (8 June 2023) and run

```
mix deps.get
mix ecto.create
```

Got problem can't compile some package (for me its cowlib) then i just run `mix deps.compile cowlib` then run `mix ecto.create` again and it work

Environment:
- Mac os ventura

<img width="808" alt="image" src="https://github.com/albirrkarim/mozilla-hubs-installation-detailed/assets/29292018/8c8ed373-79d9-4bd7-8e10-1b2f8f6b42bc">

**Comment by LuySeiwert** [comment]
Unfortunately, your suggestion didn't work, I still got the same error, but I was able to resolve this issue.

I noticed you are using Mix 1.14.4, which I found odd, as when I initially used a different elixir and erlang version than the ones prescribed I got errors saying that the elixir version specified in the .tools-version file (elixir 1.14.3) does not match the one I installed. I thought you weren't supposed to change this file as your guide says to be careful about the version of elixir and erlang.

When I tried setting the elixir version to 1.14.4 in this .tools-version file (which didn't solve anything), I saw that the specified erlang version is 23.3.4.18.  So I installed that specific version and stuck to elixir 1.14.3, which gave me a different compile error now with a hackney dependency. However, for this, I found other GitHub issues telling me to update the rebar dependency via `mix local.rebar` and that worked.

Thanks for your help, I would not have noticed the difference in erlang versions if I hadn't seen you using elixir 1.14.4!
